java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/protocol/TextDocumentServiceImpl.java

```diff
@@ -1032,7 +1032,10 @@ public CompletableFuture<List<Either<Command, CodeAction>>> codeAction(CodeActio
                             continue;
                         }
                     }
-                    Optional<Diagnostic> diag = diagnostics.stream().filter(d -> entry.getKey().equals(d.getCode().getLeft())).findFirst();
+                    Optional<Diagnostic> diag = diagnostics.stream().filter(d -> {
+                        String code = d.getCode() != null ? d.getCode().getLeft() : null;
```

Comment:
Unrelated to the core of this PR, but discovered at the same time. According to the [LSP Spec](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#diagnostic) `code` field is **optional**. Need to check for `null`.